quanta-t41/templates/flash_bmc.sh

```diff
@@ -26,12 +25,109 @@ FLASH_FILE=${DOWNLOAD_DIR}/${filename##*/}
   FLASH_FILE=${outputPath}
 <% } %>
 
+#Raw Lan command to get BMC card type
+getLanCommand=$(ipmitool raw 0x0C 0x02 0x01 0xFF 0x00 0x00)
+echo $getLanCommand
+array2=(${getLanCommand//$'\n'/ })
+echo ${array2[0]}
+bmcCardType=${array2[1]}
+echo $bmcCardType
+
+#Lan print command to get IP source info "Static Address"
+lanPrintInfo=$(ipmitool lan print)
+array3=(${lanPrintInfo//$'\n'/ })
+ipSrcInfo=${array3[54]}
+echo "${ipSrcInfo,,}"
+ipaddrInfo=${array3[59]}
+echo $ipaddrInfo
+netMaskInfo=${array3[63]}
+echo $netMaskInfo
+defGatewayInfo=${array3[100]}
+echo $defGatewayInfo
+
 #
 # Retrieve the update script and run it
 curl --retry 3 <%=api.server%>/<%= sku.bmcFirmware.command %> -o ${CMD}
 pushd ${DOWNLOAD_DIR}
 chmod 777 ${CMD}
 ${CMD} <%= sku.bmcFirmware.args %>
 
+sleep 5
+
+counter=90
+until [ $counter -le 0 ]; do
+    set +e
+    check=$(sudo ipmitool mc info | grep -c 'Manufacturer ID')
+    set -e
+    if [ $check == 1 ]; then
+        #Raw LAN command to set the BMC card type
+        ipmitool raw 0x0C 0x01 0x01 0xFF $bmcCardType
+        sleep 8
```

Comment:
After the command was accepted by the BMC it still took some time to actually occur and be committed.  Without the sleep the next command would read back the old value rather than the accepted value.